### ISSUE TITLE
New developer focused workflow article about how the login and registration works in WordPress.

### ISSUE THREAD

**Issue by bph** [WordPress/developer-blog-content#165]
### Discussed in https://github.com/WordPress/developer-blog-content/discussions/159

<div type='discussions-op-text'>

<sup>Originally posted by **Rajinsharwar** August 24, 2023</sup>
When developing user authentication systems for WordPress, it's really important to have a deep understanding of the login and registration processes. I am proposing a new article that will shed some light on the intricacies of these processes, and the underlying functions that ensure secure and seamless user experiences in WordPress sites. 

I have also created a draft documentation for the purpose. Requesting your opinion and review on this:
https://docs.google.com/document/d/17N4uVctGa1pk8B2AcqY7KGGlY6yVV9azn85TsGJGALA/edit?usp=drivesdk</div>

**Comment by bph** [comment]
[Rajinsharwar](https://github.com/Rajinsharwar) The topic was approved in today's Editorial Group Meeting. 
To assign the issue to you, you would first need to comment here.  (GitHub is funny that way)
Thanks for suggesting the topic..

**Comment by Rajinsharwar** [comment]
Hi @bph, great to hear that! Looking forward to get this assigned so that I can include more code references, and proof-read the doc.

**Comment by Rajinsharwar** [comment]
Hi @bph, I just finished adding more code examples, and more proof-reads. Requesting a review on this: 🙂

https://docs.google.com/document/d/17N4uVctGa1pk8B2AcqY7KGGlY6yVV9azn85TsGJGALA/edit?usp=sharing

**Comment by marybaum** [comment]
On it! Before Monday in the US

**Comment by bph** [comment]
@Rajinsharwar to give reveiwers access for comments, could you please make the Google Doc accessible for comments via link?

**Comment by Rajinsharwar** [comment]
Done, updated the access @bph :)

**Comment by bph** [comment]
@Rajinsharwar I took a first look but need to take a break now.  I will get back to it sometime tomorrow. Sorry. maybe @marybaum or @justintadlock are available for a review.

**Comment by bph** [comment]
Milana @zzap has volunteered to do a second review on your post.

**Comment by Rajinsharwar** [comment]
Great! Will be waiting for his feedback on this :)

**Comment by zzap** [comment]
Hello @Rajinsharwar , I have reviewed your draft and left quite a few comments. First of all, thank you very much for taking the time to write this article. It is an important topic and you certainly spent a lot of time working on it.

The majority of code examples need `wporg` prefix, as it is a common prefix for examples on wp.org. It's also one of the best practices for coding with WordPress and we want to enforce that. 

Some code examples are missing internationalisation functions, and some only textdomain. That as well is expected from all code examples living on wp.org. 

One callback, however, is causing Fatal error. I didn't go into detailed debugging, but maybe some hint would be that you are passing 5 arguments but not actually using them. 

I don't know how much experience you have in writing software documentation, but I have a feeling that you are going the extra mile to prevent content from being dry or repeated. This is okay, but be careful not to lose the point and clarity while doing so. I commented on a few places which crossed this line and became unclear and difficult to understand or simply too novelistic. With software documentation, be it tutorial or code reference, it's better to be dry but clear than to lose the point in an attempt to be more interesting.

Thank you, again, so much for your time and commitment. I'm looking forward to seeing this article published.

**Comment by bph** [comment]
@Rajinsharwar I was wondering on the progress of this article. Are there any blockers I can remove?

**Comment by abhansnuk** [comment]
Following the editorial meeting, following up on this article see if any help is needed to move this forward. Thanks to Rajen and all the reviewers for all your time and effort on this so far.

**Comment by Rajinsharwar** [comment]
Hi @bph @abhansnuk, hope you're doing great! Sorry for the delay, I have fixed the issues commented on by @zzap 😃. I hope this is ready for a second review! Let me know if I missed any step!

Thanks to @zzap as well for the detailed feedback, yep, I am new to software documentation, so I will of course make sure to follow your suggestions in future cases. Thanks again, and have a great day! 🙂

**Comment by bph** [comment]
@abhansnuk would you have time to give this a 2nd review? [Google Doc](https://docs.google.com/document/d/17N4uVctGa1pk8B2AcqY7KGGlY6yVV9azn85TsGJGALA/edit)

**Comment by abhansnuk** [comment]
Hello, I have done the second review and checked in following along on the two items I raised. This is ready. 

It needs an excerpt adding, contributors names, and any tags etc. 

Great team work everyone! Can’t wait to see this live. Hope you will join us for another post soon Rajin.

**Comment by bph** [comment]
@Rajinsharwar seems your article is about ready to publish. 

So we need to get you access to the site so you can post in draft. I sent you an invite to the site. You need to accept it and you are off to the races :-) 

### Pre-publishing checklist: (updated 1/29/2024)

- [x] Post Title and subheaders in sentence case
- [x] Are Category or Categories selected?
- [x] Are Tags identifies?
- [x] Is there an explicit Excerpt?
- [x] Are all images files uploaded to the media library
- [x] Do all images have an alt-text?
- [x] Assign or upload a featured image
- [x] Props added? (See [Guidelines](https://make.wordpress.org/core/handbook/best-practices/post-comment-guidelines/#giving-proper-credit-props))
- [x] add **copy for a social post** as comment to this issue ([example](https://github.com/WordPress/developer-blog-content/issues/152#issuecomment-1654259045))
🙌 Publish! 📗

### Post-publishing checklist
- [x] add Props for reviews to #props channel in WP Slack ([Example](https://wordpress.slack.com/archives/C0FRG66LR/p1689098122307029)) (use Slack handles) 
- [x] add 'post on social' label to the issue
- [x] close the issue with a comment to link to the published post
- [x] close the accompanying discussion with the link to the published post.

**Comment by bph** [comment]
Please notify me when you are done with the pre-publish check list. As this is your very first post, please set a Public Preview link and share it here so I can double-check it 
![Screenshot 2024-02-08 at 15 44 49](https://github.com/WordPress/developer-blog-content/assets/39980/be7a4177-e013-451c-9f50-f22cf19e8c25)

**Comment by Rajinsharwar** [comment]
Hi @bph @abhansnuk and all, thank you so much for all your help! I made the post available for public preview: https://developer.wordpress.org/news/?p=2702&preview=1&_ppp=a1d48c5d2b

Social copy:

Gain developer-level insights into the backend functions of user authentication in WordPress with a focus on extending and modifying each step:

[Post link after publishing]

**Comment by bph** [comment]
The post is now published after I reviewed the post and made the following changes: 

- Code blocks need to have a language attribution for color coding

```<code lang="php" class="language-php">```
- There is a bug in the code block that add `<br/>` when adding a new line that is ignored by the `<pre>` tag, so I went through all the code blocks and added new lines again via the code editor. 
- All links: I removed the check marl for "Open in new tab", as that's not accessible (see guidelines) 
- Added all links again on the bottom as a quick way to access those
- renamed FAQ section to additional notes (no questions asked) and removed the bullet points, and added bolding, all for better readability. 
- added ticks to function and file name mentions throughout the post. 
- Shortened the excerpt, and updated the tags. 

🙌 🎉Congrats to @Rajinsharwar for the first post published on the Developer Blog!
